cloud/scope/machine.go

```diff
@@ -200,27 +200,33 @@ func (m *MachineScope) GetOrCreateMachine(ctx context.Context) (*core.Instance,
 	}
 
 	failureDomain := m.Machine.Spec.FailureDomain
-	if failureDomain == nil {
-		m.Logger.Info("Failure Domain is not set in the machine spec, setting it to a random value from 1 to 3")
-		randomFaultDomain, err := rand.Int(rand.Reader, big.NewInt(3))
+	var faultDomain string
+	var availabilityDomain string
+	if failureDomain != nil {
+		failureDomainIndex, err := strconv.Atoi(*failureDomain)
 		if err != nil {
-			m.Logger.Error(err, "Failed to generate random fault domain")
+			m.Logger.Error(err, "Failure Domain is not a valid integer")
+			return nil, errors.Wrap(err, "invalid failure domain parameter, must be a valid integer")
+		}
+		m.Logger.Info("Failure Domain being used", "failure-domain", failureDomainIndex)
+		if failureDomainIndex < 1 || failureDomainIndex > 3 {
+			err = errors.New("failure domain should be a value between 1 and 3")
+			m.Logger.Error(err, "Failure domain should be a value between 1 and 3")
+			return nil, err
+		}
+		faultDomain = m.OCIClusterAccessor.GetFailureDomains()[*failureDomain].Attributes[FaultDomain]
+		availabilityDomain = m.OCIClusterAccessor.GetFailureDomains()[*failureDomain].Attributes[AvailabilityDomain]
+	} else {
+		randomFailureDomain, err := rand.Int(rand.Reader, big.NewInt(3))
```

Comment:
so I understood https://github.com/oracle/cluster-api-provider-oci/issues/351 to be requesting the ability to not set the FD at all and let compute figure out the FD. 

> We'd like to know if it is feasible to leave FD unassigned

The way I read this is `if` allows the user to set the FD, the `else` will randomly assign. But I think they want a third option.

Reply:
The requirement, we clarified, is to not set a random FD if user has not set the FD. AD is a mandatory param. So if fault domain is not set, then we will set a random AD, and FD wont be set. In the case of PCA, thew wont set faulire domain for cases where user does not specify them, and hence it will go ti the else block.

Reply:
But good point, I will edit the ticket to clarify.